app/src/main/java/org/listenbrainz/android/ui/screens/yim23/Yim23StatsHeatMapScreen.kt

```diff
@@ -0,0 +1,164 @@
+            Text("(${mostListenedMonth.second} Songs)" , textAlign = TextAlign.Center , color = MaterialTheme.colorScheme.background , style = MaterialTheme.typography.bodyLarge)
+            Yim23HeatMap(viewModel = viewModel)
+            Spacer(modifier = Modifier.padding(bottom = 11.dp))
+            Yim23Footer(footerText = "MY STATS", isUsername = true, navController = navController, downScreen = Yim23Screens.YimStatsGraphScreen)
+        }
+    }
+}
+
+@Composable
+private fun Yim23HeatMap (viewModel: Yim23ViewModel , paddings: YimPaddings = LocalYimPaddings.current) {
+    Box(
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(start = 11.dp, end = 11.dp)
+            .clip(RoundedCornerShape(10.dp))
+            .height(240.dp)
+            .background(Color(0xFFe0e5de))
+            .padding(
+                start = paddings.defaultPadding,
+                end = paddings.defaultPadding,
+                bottom = paddings.defaultPadding
+            ),
+
+    ) {
+
+        Column(
+            Modifier
+                .fillMaxSize()
+                .padding(horizontal = paddings.smallPadding),
+            verticalArrangement = Arrangement.Center
+        ) {
+
+            // Month row
+            val listState = rememberLazyListState()
+            LazyRow(
+                state = listState,
+                modifier = Modifier
+                    .height(20.dp),
+                userScrollEnabled = false
+            ){
+                items(listOf("Jan","Feb","Mar","Apr","May","Jun","Jul","Aug","Sep","Oct","Nov","Dec")
+                ){ month ->
+                    Text(
+                        text = month,
+                        modifier = Modifier
+                            .width(89.25.dp),
+                        textAlign = TextAlign.Center,
+                        fontSize = 12.sp
+                    )
+                }
+            }
+            val gridState = rememberLazyGridState()
+
+            LaunchedEffect(gridState.isScrollInProgress){
+                listState.animateScrollToItem(index = gridState.firstVisibleItemIndex/26)
+            }
+            // Heat Map Grid
+            val listensOfYear = viewModel.getListensListOfYear()
+            LazyHorizontalGrid(
+                state = gridState,
+                modifier = Modifier
+                    .height(160.dp)
+                    .padding(bottom = paddings.smallPadding),
+                rows = GridCells.Fixed(7)
+            ){
+                items( listensOfYear )
+                { item ->
+
+                    // Heatmap square
+                    Box(
+                        modifier = Modifier
+                            .width(20.dp)
+                            .padding(1.dp)
+                            .background(
+                                when {
+                                    item >= 150 -> Color(0xFFE5743E)
```

Comment:
If the colors are different for YIM 22 and 23, we can pass a function which outputs a color as follows: `fun determineColor(value: Int)`